modules/react/common/stories/Responsive.stories.mdx

```diff
@@ -0,0 +1,241 @@
+
+### Container vs Viewport
+
+There are two ways we will look at responsive styling below. They are container-based and
+viewport-based responsive styling.
+
+> Using container-based responsive styling, we will measure the size of the container the element is
+> in and apply styles based on the size of that container.
+
+> Using viewport-based responsive styling, we will measure the size of the user's viewport and apply
+> styles based on the size of that viewport.
+
+If you are measuring a user's viewport size, then you should be using viewport-based responsive
+styling.
+
+If you only want to measure the size of a container, then you should be using container-based
+responsive styling.
+
+> **Note**: A container's size can be affected by surrounding containers or the viewport size.
+
+### Canvas Breakpoints
+
+In Canvas Kit, there are a set of breakpoints:
+
+| Zero  | Small   | Medium  | Large    | Extra Large |
+| ----- | ------- | ------- | -------- | ----------- |
+| > 0px | > 320px | > 768px | > 1024px | > 1440px    |
+
+> Zero: This is applied from 0 - 319px
+
+> Small: This is applied from 320px - 767px
+
+> Medium: This is applied from 768px - 1023px
+
+> Large: This is applied from 1024px - 1439px
+
+> Extra Large: This is applied from 1440px and up
+
+These are the breakpoints we will use when we apply styles based on the size of a container or
+viewport.
+
+## Container-based Responsive Styling
+
+In the interactive example below, when the dropdown's value is updated, the container size will
+change to match the size shown in the dropdown.
+
+<ExampleCodeBlock code={ResponsiveContainer} />
+
+This example uses the following hooks:
+
+- `useResizeObserver` provides the width (and height) of the container.
+
+- `useResponsiveContainerStyles` allows you to create container-based responsive styles with
+  objects. This hook accepts three arguments: style objects, container width, and theme (optional).
+  Each style object accepts keys corresponding to the following sizes: `zero`, `s`, `m`, `l` and
+  `xl`. These sizes represent [Canvas breakpoints](#canvas-breakpoints). The sizes act like
+  `min-width`. For example, if you want to apply styles from `medium` and up, then you would write
+  those styles under `m`.
+
+## Viewport-based Responsive Styling
+
+The viewport is the user's visible area of a web page (varies with screen size).
+
+We can use media queries to run a series of tests that will measure the screen size or resolution
+and apply CSS styles specifically to an element.
+
+```css
+@media screen and (min-width: 800px) {
+  .container {
+    margin: 1em 2em;
+  }
+}
+```
+
+In the interactive example below, styles are applied based on the size of the user's viewport using
+our breakpoint utilities: `up()`, `down()`, `between()` and `only()`. Note the use of `up()` and
+`down()` near the top of the example code. Adjust the width of your browser to see the layout adjust
+to the available width.
```

Comment:
I'm removing the parentheses from `up()` and `down()`. We don't typically use parentheses when referencing functions in our documentation though I'm wondering if we should to make it more obvious we're referring to a function. Something to think about for the future.

```suggestion
two of our breakpoint utilities: `up` and `down`. Adjust the width of your browser to see the layout
adjust to the available width.
```